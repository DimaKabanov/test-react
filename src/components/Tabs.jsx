import React, { Component } from 'react';
import 'react-tabs/style/react-tabs.css';
import {
  Tabs as TabsWrapper,
  Tab,
  TabList,
  TabPanel,
} from 'react-tabs';

class Tabs extends Component {
  renderTab = tabs => tabs.map(tab => <Tab data-tab-id={`tab-${tab.id}`} key={tab.id}>{tab.title}</Tab>)

  renderTabPanel = tabs => tabs.map(tab => <TabPanel key={tab.id}>{tab.text}</TabPanel>)

  render() {
    const { tabs } = this.props;
    return (
      <TabsWrapper>
        <TabList>
          {this.renderTab(tabs)}
        </TabList>
        <React.Fragment>
          {this.renderTabPanel(tabs)}
        </React.Fragment>
      </TabsWrapper>
    );
  }
}

export default Tabs;
