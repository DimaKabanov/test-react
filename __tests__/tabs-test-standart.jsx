import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from '../src/components/Tabs';

const tabsData = [
  {
    id: 0,
    title: 'Mario',
    text: 'Mario ipsum dolor sit, amet consectetur adipisicing elit.',
  },
  {
    id: 1,
    title: 'Peach',
    text: 'Peach ratione laudantium inventore ullam debitis optio rerum eum modi.',
  },
  {
    id: 2,
    title: 'Yoshi',
    text: 'Yoshi animi odit officia temporibus est enim, quas at facilis officiis? Itaque!',
  },
];

test('Tabs 1', () => {
  const component = renderer.create(<Tabs tabs={tabsData} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
