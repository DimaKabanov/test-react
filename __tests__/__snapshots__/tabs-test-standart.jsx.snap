// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs 1 1`] = `
<div
  className="react-tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      className="react-tabs__tab react-tabs__tab--selected"
      data-tab-id="tab-0"
      id="react-tabs-0"
      role="tab"
      tabIndex="0"
    >
      Mario
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      className="react-tabs__tab"
      data-tab-id="tab-1"
      id="react-tabs-2"
      role="tab"
      tabIndex={null}
    >
      Peach
    </li>
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="false"
      className="react-tabs__tab"
      data-tab-id="tab-2"
      id="react-tabs-4"
      role="tab"
      tabIndex={null}
    >
      Yoshi
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    Mario ipsum dolor sit, amet consectetur adipisicing elit.
  </div>
  <div
    aria-labelledby="react-tabs-2"
    className="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-4"
    className="react-tabs__tab-panel"
    id="react-tabs-5"
    role="tabpanel"
  />
</div>
`;
